Mathlib/RingTheory/AdjoinRoot.lean

```diff
@@ -334,6 +338,26 @@ theorem noZeroSMulDivisors_of_prime_of_degree_ne_zero [IsDomain R] (hf : Prime f
 
 end Prime
 
+section
+
+/-- If `f = g`, `R` adjoint a root of `f` is isomorphic to `R` adjoint a root of `g`. -/
+def algEquivOfEq {f g : R[X]} (hfg : f = g) : AdjoinRoot f ≃ₐ[R] AdjoinRoot g :=
+  .ofAlgHom
+    (liftHom f (root g) (by simp [hfg]))
+    (liftHom g (root f) (by simp [hfg]))
+    (by ext; simp)
+    (by ext; simp)
+
+variable {f g : R[X]} (hfg : f = g)
+
+@[simp] lemma algEquivOfEq_root : algEquivOfEq hfg (root f) = root g := by
+  simp [algEquivOfEq]
+
+@[simp] lemma algEquivOfEq_symm_root : (algEquivOfEq hfg).symm (root g) = root f := by
+  simp [algEquivOfEq]
```

Comment:
The right simp lemma might be `(algEquivOfEq hfg).symm = algEquivOfEq hfg.symm` instead.